lib/std/mem.zig

```diff
@@ -876,6 +874,63 @@ pub fn indexOfPos(comptime T: type, haystack: []const T, start_index: usize, nee
     return null;
 }
 
+fn boyerMooreHorspoolPreprocess(pattern: []const u8, table: []usize) void {
+    for (table) |*c| {
+        c.* = pattern.len;
+    }
+
+    var i: usize = 0;
+    while (i < pattern.len - 1) : (i += 1) {
+        table[pattern[i]] = pattern.len - 1 - i;
+    }
+}
+/// Find the index in a slice of a sub-slice, searching from the end backwards.
+/// To start looking at a different index, slice the haystack first.
+// Reverse boyer-moore-horspool algorithm
+pub fn lastIndexOf(comptime T: type, haystack: []const T, needle: []const T) ?usize {
+    if (haystack.len < 32 or needle.len <= 2)
+        return lastIndexOfNaive(T, haystack, needle);
+
+    if (needle.len > haystack.len or needle.len == 0) return null;
+
+    const haystackU8 = @bitCast([]const u8, haystack);
+    const needleU8 = @bitCast([]const u8, needle);
+
+    var table: [256]usize = undefined;
+    boyerMooreHorspoolPreprocess(needleU8, table[0..]);
+
+    var i: usize = 0;
+    while (i <= haystackU8.len - needleU8.len) {
+        const reverseIndex = haystackU8.len - i - needleU8.len - 1;
+        if (mem.eql(u8, haystackU8[reverseIndex .. reverseIndex + needleU8.len], needleU8)) return i;
```

Comment:
Returning `i` is wrong as that's the index starting from the left.

Reply:
fixed this function now..